Source/WebKit/WebProcess/haiku/WebProcessMainHaiku.cpp

```diff
@@ -39,13 +39,15 @@ class WebProcessMain final : public AuxiliaryProcessMainBase {
 public:
     bool platformInitialize() override
     {
-        BApplication* app = new BApplication("application/x-vnd.haiku-webkit.webprocess");
```

Comment:
You really need a BApplication to do anything. Where is it created, if not here?

Reply:
The ui process is a bapplication right wont that suffice?

Reply:
No. Without the BApplication, 1) the worker will have no link to the app_server for things like drawing ops, and 2) It won't be registered with the roster, and consequently you'd have no way of sending e.g. BMessages between it and the UI process.

Reply:
I see so that explains why my messages were not recieved

Reply:
so this will be my main run loop that gets all my messages right?

Reply:
For inter-process communications, yes.

Reply:
Yes. I think the runloop implementation detects that it is in the main thread, and in that cases it will run the BApplication looper (using be_app->Run()). Otherwise (for various worker threads) it will create its own worker instead.